webpack/assets/javascripts/react_app/components/common/Slot/SlotSelectors.js

```diff
@@ -3,7 +3,7 @@ import { getSlotComponents } from '../../../../services/SlotsRegistry';
 export const selectComponentByWeight = slotId =>
   getSlotComponents(slotId)
     .sort((a, b) => b.weight - a.weight)
-    .map(c => c.component);
+    .map(c => c.component) || {};
```

Comment:
feels like we could use `createSelector` from `reselect` here

Reply:
the components aren't stored in redux though, but in a special registry so memorization isn't effective here.